### PR TITLE
Add code coverage reporting

### PR DESCRIPTION
**Pull request by ssbarnea**
Partial: #52

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -322,6 +322,14 @@ jobs:
     - name: Run testing against the Git checkout
       run: npm test
 
+    - name: Produce coverage report
+      run: npm run coverage
+    - name: Upload coverage data
+      uses: codecov/codecov-action@v2
+      with:
+        name: ${{ matrix.name }}
+        files: out/coverage.lcov
```
Comment (webknjaz) [inline .github/workflows/ci.yml:331]: Let's flag each report. This way we can see which code was executed on each platform.
```suggestion
        files: out/coverage.lcov
        flags: >-
          GHA,
          ${{ runner.os }},
          ${{ matrix.os }},
          ${{ matrix.node-version }}
```

File: package.json [hunk 2]
```diff
@@ -62,9 +62,11 @@
     "@typescript-eslint/eslint-plugin": "^4.28.3",
     "@typescript-eslint/parser": "^4.28.3",
     "chai": "^4.3.4",
+    "codecov": "^3.8.3",
```
Comment (webknjaz) [inline package.json:65]: Is this really needed? Is it that deprecated uploader? It won't be used for uploading, the action has its own uploader (the new one).

File: package.json [hunk 3]
```diff
@@ -62,9 +62,11 @@
...
     "eslint": "^7.32.0",
     "eslint-config-prettier": "^8.3.0",
     "mocha": "^8.4.0",
+    "nyc": "^15.1.0",
```
Comment (webknjaz) [inline package.json:69]: I'm pretty sure this is the only thing that is needed in the deps to collect the coverage. Not codecov. Anything I'm missing?

File: codecov.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+codecov:
+  require_ci_to_pass: true
```
Comment (webknjaz) [inline codecov.yml:2]: Let's do this to reduce the flakiness:
```suggestion
  notify:
    after_n_builds: 8  # the number of individual jobs in the test matrix
    wait_for_ci: yes
  require_ci_to_pass: true
```

**Review by webknjaz — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (ssbarnea) [reply package.json]: Right, with action we don't really need it.

**Pull request merged**